packages/link-modules/src/repositories/pivot.ts

```diff
@@ -0,0 +1,104 @@
+import { Context, FindOptions, ModuleJoinerConfig } from "@medusajs/types"
+import {
+  EntitySchema,
+  LoadStrategy,
+  FilterQuery as MikroFilterQuery,
+  FindOptions as MikroOptions,
+} from "@mikro-orm/core"
+
+import {
+  MikroOrmAbstractBaseRepository,
+  generateEntityId,
+} from "@medusajs/utils"
+import { SqlEntityManager } from "@mikro-orm/postgresql"
+
+export function getPivotRepository(model: EntitySchema) {
+  return class PivotRepository extends MikroOrmAbstractBaseRepository<object> {
+    readonly manager_: SqlEntityManager
+    readonly model_: EntitySchema
+    readonly joinerConfig_: ModuleJoinerConfig
+
+    constructor({
+      manager,
+      joinerConfig,
+    }: {
+      manager: SqlEntityManager
+      joinerConfig: ModuleJoinerConfig
+    }) {
+      // @ts-ignore
+      super(...arguments)
+      this.manager_ = manager
+      this.model_ = model
+      this.joinerConfig_ = joinerConfig
+    }
+
+    async find(
+      findOptions: FindOptions<unknown> = { where: {} },
+      context: Context = {}
+    ): Promise<unknown[]> {
+      const manager = this.getActiveManager<SqlEntityManager>(context)
+
+      const findOptions_ = { ...findOptions }
+      findOptions_.options ??= {}
+
+      Object.assign(findOptions_.options, {
+        strategy: LoadStrategy.SELECT_IN,
+      })
+
+      return await manager.find(
+        this.model_,
+        findOptions_.where as MikroFilterQuery<unknown>,
+        findOptions_.options as MikroOptions<any>
+      )
+    }
+
+    async findAndCount(
+      findOptions: FindOptions<object> = { where: {} },
+      context: Context = {}
+    ): Promise<[object[], number]> {
+      const manager = this.getActiveManager<SqlEntityManager>(context)
+
+      const findOptions_ = { ...findOptions }
+      findOptions_.options ??= {}
+
+      Object.assign(findOptions_.options, {
+        strategy: LoadStrategy.SELECT_IN,
+      })
+
+      return await manager.findAndCount(
+        this.model_,
+        findOptions_.where as MikroFilterQuery<unknown>,
+        findOptions_.options as MikroOptions<any>
+      )
+    }
+
+    async delete(data: any, context: Context = {}): Promise<void> {
```

Comment:
question: Shouldn't the pivot table be soft deleted for compensation?

Reply:
They are softDeleted. But the softDelete methods is implemented at the baseClass on the utils package.
This `delete` here is just to have them per convention in all modules.